.github/workflows/pre-commit.yml

```diff
@@ -0,0 +1,42 @@
+name: pre-commit
+
+on:
+  pull_request:
+  push:
+    branches: [master]
+
+jobs:
+  pre-commit:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v3
+
+    - uses: ./.github/actions/setup-build
+      with:
+        ghc: "9.2.3"
+        os: ${{ runner.os }}
```

Comment:
You should not shorten the project name for this, I believe. (Otherwise format can change)

Reply:
Fixed!